### PR TITLE
Add tests for the source model 

### PR DESCRIPTION
**Pull request by adivijaykumar**
Adding tests.

### CODE REVIEW

File: test/gw/detector/interferometer_test.py [hunk 1]
```diff
@@ -161,6 +161,57 @@ def test_get_detector_response_multiple_modes(self):
...
         )
 
+    def test_get_detector_response_relative_binning_default_behaviour(self):
+        self.ifo.antenna_response = MagicMock(return_value=1)
+        self.ifo.time_delay_from_geocenter = MagicMock(return_value=0)
+        self.ifo.epoch = 0
+        self.minimum_frequency = 10
+        self.maximum_frequency = 20
+        plus = np.linspace(0, 4096, 4097)
+        response = self.ifo.get_detector_response(
```
Comment (krishnakruthi) [inline test/gw/detector/interferometer_test.py]: Shouldn't this be `get_detector_response_relative_binning`?

File: test/gw/detector/interferometer_test.py [hunk 2]
```diff
@@ -161,6 +161,57 @@ def test_get_detector_response_multiple_modes(self):
...
+        )
+
+    def test_get_detector_response_relative_binning_with_dt(self):
+        self.ifo.antenna_response = MagicMock(return_value=1)
+        self.ifo.time_delay_from_geocenter = MagicMock(return_value=0)
+        self.ifo.epoch = 1
+        self.minimum_frequency = 10
+        self.maximum_frequency = 20
+        plus = np.linspace(0, 4096, 4097)
+        response = self.ifo.get_detector_response(
```
Comment (krishnakruthi) [inline test/gw/detector/interferometer_test.py]: Same thing here

File: test/gw/detector/interferometer_test.py [hunk 3]
```diff
@@ -161,6 +161,57 @@ def test_get_detector_response_multiple_modes(self):
...
+
+    def test_get_detector_response_relative_binning_multiple_modes(self):
+        self.ifo.antenna_response = MagicMock(return_value=1)
+        self.ifo.time_delay_from_geocenter = MagicMock(return_value=0)
+        self.ifo.epoch = 0
+        self.minimum_frequency = 10
+        self.maximum_frequency = 20
+        plus = np.linspace(0, 4096, 4097)
+        cross = np.linspace(0, 4096, 4097)
+        response = self.ifo.get_detector_response(
```
Comment (krishnakruthi) [inline test/gw/detector/interferometer_test.py]: Here too!

**Review by krishnakruthi — APPROVED** [review]

File: test/gw/detector/interferometer_test.py (hunk 1 above)
Comment (adivijaykumar) [reply test/gw/detector/interferometer_test.py]: Stupid me, I don't think I made any changes to these methods. Sorry!

**Comment by adivijaykumar** [comment]
@Kruthi24 I have actually got rid of all the interferometer tests since I think it would require some more thought. Right now, this PR only adds tests for the source model. Do you think this is ready for merge in that case?

**Comment by krishnakruthi** [comment]
Yup!

**Pull request merged**